### ISSUE TITLE
Solo-mining --no-getwork

### ISSUE THREAD

**Issue by khaotekmetalcraft** [KlausT/ccminer#226]
+1 on the --no-getwork / coinbase addy + stale share submit

I use your software every day, using it less and less because of inability to solo on my rigs.

Needs to be incorporated asap especially considering profits are down so far and it's becoming less and less feasible to throw money away at a pool when there's a significant chance for 100% profits.

Also, I know there's a couple rips on it already but your neoscrypt algo really needs a rework.  Pulling significantly less than HSRminer-neo on the 1070/80 lineup to the tune of 600+ hashes when run in mixed card groups, also individually.  If you'd like screens/+bios make model, lmk.  Have 1080ti/x80/x70ti/x70/x60.3g/ also 980 maxwell groups.

-khao

**Comment by KlausT** [comment]
Yeah, I'm thinking about adding getblocktemplate solo mining in the near future.
What would be your favourite coin to solo mine?

https://github.com/KlausT/ccminer/issues/150

**Comment by khaotekmetalcraft** [comment]
Don't know why but I get a huge charge out of watching Neoscrypt spam on Feathercoin, amusing to watch the hash spikes from NH on the diff dips.  Right now my favorite algos are Lyra2z (zcoin) (high frequency very low power), Neoscrypt (feather) and Lyra2Rev2 (Vert).  X16R is pretty damned slick to, gotta give em credit.  I like the variable algo and general instability premise.  +1 Negate the need for over clocking, Force low power and default/lower clocks and level the playing field.  And fuck ASIC.

**Comment by khaotekmetalcraft** [comment]
All time favorite would be Equihash.  Was a huge zCash fan / proponent until I learned / took note of it's inclusion into the banking industry & also their (and others like them) abuse towards BTCP on twitter.  Haters always gonna hate though.  I say: Privacy Coins, unite.  Fuck the system.  Encrypt, decentralize and take no prisoners.

**Comment by khaotekmetalcraft** [comment]
What's your favorite coin/algo?

**Comment by tigeratom59** [comment]
Thank

**Comment by lordgamegenie** [comment]
Has solo mining been added to the latest release.  Really need to figure out how to mine these obscure and unknown alt-coin wallets I set up, and the least hassle way I can think of doing this is using the --coinbase-addr function.

**Comment by KlausT** [comment]
Some wallets still support the getwork command. It should be no problem solomining these coins.
Like Orbitcoin for example:
`ccminer -O rpcuser:rpcpassword -o http://localhost:15299 -a neoscrypt --no-stratum`

I have not implemented the getblocktemplate command yet, Many coins can't be solomined with the currect ccminer version.

**Comment by AllenChong** [comment]
👍

**Comment by ChekaZ** [comment]
Any plans to implement the --no-getwork and nTime for getblocktemplate in the near future? - Trezarcoin would need this.

**Comment by KlausT** [comment]
getblocktemplate support is actually on my todo list, but I can't promise that it will be ready soon.

**Comment by lordgamegenie** [comment]
@KlausT are you going to release a new ccminer soon? Could it support the lyra2rev3 algo?

**Comment by KlausT** [comment]
Yes, I will add lyra2rev3

**Comment by ChekaZ** [comment]
Any update on the getblocktemplate support?

**Comment by juanpc2018** [comment]
the war between GBT, getwork and stratum, very interesting differences...
there is a Miner Proxy, that accepts getwork, GBT or Stratum, and outputs Stratum...
if also could output GBT or Getwork, would be perfect, flawless...

https://slushpool.com/media/download/mining_proxy.exe
https://github.com/slush0/stratum-mining-proxy/
https://slushpool.com/help/topic/mining-proxy/
https://cryptomining-blog.com/tag/stratum-mining-proxy/
https://github.com/slush0/stratum-mining-proxy
https://mining.bitcoin.cz/media/download/mining_proxy.exe

**Comment by yukkitatsu** [comment]
hex please and x21s
